### PR TITLE
update ntp module

### PR DESCRIPTION
**Pull request by cgzones**
* add private lock type
* dontaudit sys_resource

### CODE REVIEW

File: ntp.te [hunk 1]
```diff
@@ -47,28 +52,35 @@ init_system_domain(ntpd_t, ntpdate_exec_t)
...
 
-allow ntpd_t self:capability { chown dac_override kill setgid setuid sys_time ipc_lock ipc_owner sys_chroot sys_nice sys_resource };
-dontaudit ntpd_t self:capability { net_admin sys_tty_config fsetid sys_nice };
+# sys_time : modify system time
+allow ntpd_t self:capability { chown dac_override kill setgid setuid sys_time ipc_lock ipc_owner sys_chroot sys_nice };
+dontaudit ntpd_t self:capability { net_admin sys_tty_config fsetid sys_nice sys_resource };
 allow ntpd_t self:process { signal_perms getcap setcap setsched setrlimit };
 allow ntpd_t self:fifo_file rw_fifo_file_perms;
 allow ntpd_t self:shm create_shm_perms;
+allow ntpd_t self:socket create;
```
Comment (pebenito) [inline ntp.te]: What type of socket is this?

File: ntp.te [hunk 2]
```diff
@@ -47,28 +52,35 @@ init_system_domain(ntpd_t, ntpdate_exec_t)
...
 manage_dirs_pattern(ntpd_t, ntp_drift_t, ntp_drift_t)
 manage_files_pattern(ntpd_t, ntp_drift_t, ntp_drift_t)
 files_var_filetrans(ntpd_t, ntp_drift_t, file)
 
-allow ntpd_t ntp_conf_t:file read_file_perms;
-
 read_files_pattern(ntpd_t, ntpd_key_t, ntpd_key_t)
 read_lnk_files_pattern(ntpd_t, ntpd_key_t, ntpd_key_t)
 
+allow ntpd_t ntpd_lock_t:file write_file_perms;
```
Comment (pebenito) [inline ntp.te]: It never creates or deletes its own lock file?

File: ntp.te (hunk 1 above)
Comment (cgzones) [inline ntp.te]: I'll investigate

File: ntp.te (hunk 2 above)
Comment (cgzones) [inline ntp.te]: the lockfile is created by systemd `init_daemon_lock_file(ntpd_lock_t, file, "ntpdate")`, and I did not encounter any unlink request yet?!

File: ntp.te (hunk 1 above)
Comment (cgzones) [inline ntp.te]:
```
type=PROCTITLE msg=audit(02/17/17 22:52:21.790:167) : proctitle=/usr/sbin/ntpd -p /var/run/ntpd.pid -g -u 106:111 
type=SYSCALL msg=audit(02/17/17 22:52:21.790:167) : arch=armeb syscall=socket per=PER_LINUX_32BIT success=no exit=EAFNOSUPPORT(Address family not supported by protocol) a0=unknown family(0x0) a1=SOCK_DGRAM a2=ip a3=0x48381b00 items=0 ppid=1 pid=540 auid=unset uid=ntp gid=ntp euid=ntp suid=ntp fsuid=ntp egid=ntp sgid=ntp fsgid=ntp tty=(none) ses=unset comm=ntpd exe=/usr/sbin/ntpd subj=system_u:system_r:ntpd_t:s0 key=(null) 
type=AVC msg=audit(02/17/17 22:52:21.790:167) : avc:  denied  { module_request } for  pid=540 comm=ntpd kmod="net-pf-0" scontext=system_u:system_r:ntpd_t:s0 tcontext=system_u:system_r:kernel_t:s0 tclass=system permissive=1 
type=AVC msg=audit(02/17/17 22:52:21.790:167) : avc:  denied  { create } for  pid=540 comm=ntpd scontext=system_u:system_r:ntpd_t:s0 tcontext=system_u:system_r:ntpd_t:s0 tclass=socket permissive=1 
```

Comment (cgzones) [inline ntp.te]: i opened a bugreport: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=855444

**Pull request merged**